includes/batch.inc

```diff
@@ -113,8 +113,20 @@ function _drush_backend_batch_process($command = 'batch-process', $args = [], $o
         $batch['id'] = $batch_storage->getId();
       }
       catch (IntegrityConstraintViolationException) {
-        // Remove calls to nextId() once 10.0 is no longer supported.
-        $batch['id'] = \Drupal::database()->nextId();
+        // @todo this is here to support the update path to deprecate
+        // Connection::nextId(). Remove when Drupal 10.1 and lower is no longer
+        // supported.
```

Comment:
What exactly should be removed after 10.1 is unsupported. We dont need to catch the exception at all anymore?

Reply:
correct. the exception is thrown when the 10.1 => 10.2 update function hasn't been run yet, because the id field is not a serial/autoincrement yet. (the update function is incorrectly named system_update_10100(), it should have been system_update_10200()